pages/auth/forgot-password/index.tsx

```diff
@@ -0,0 +1,153 @@
+import Head from "next/head";
+import React from "react";
+import { getCsrfToken } from "next-auth/client";
+import debounce from "lodash.debounce";
+
+export default function Page({ csrfToken }) {
+  const [loading, setLoading] = React.useState(false);
+  const [error, setError] = React.useState(null);
+  const [success, setSuccess] = React.useState(false);
+  const [email, setEmail] = React.useState("");
+
+  const handleChange = (e) => {
+    setEmail(e.target.value);
+  };
+
+  const submitForgotPasswordRequest = async ({ email }) => {
+    try {
+      const res = await fetch("/api/auth/forgot-password", {
+        method: "POST",
+        body: JSON.stringify({ email: email }),
+        headers: {
+          "Content-Type": "application/json",
+        },
+      });
+
+      const json = await res.json();
+      if (!res.ok) {
+        setError(json);
+      } else {
+        setSuccess(true);
+      }
+
+      return json;
+    } catch (reason) {
+      setError({ message: "An unexpected error occurred. Try again." });
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const debouncedHandleSubmitPasswordRequest = debounce(submitForgotPasswordRequest, 250);
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (!email) {
+      return;
+    }
+
+    if (loading) {
+      return;
+    }
+
+    setLoading(true);
+    setError(null);
+    setSuccess(false);
+
+    await debouncedHandleSubmitPasswordRequest({ email });
+  };
+
+  const Success = () => {
+    return (
+      <div className="space-y-6">
```

Comment:
Minor point; my initial impression was that the spacing was too much here, `space-y-4` I think is more suitable. That said; just my gut feeling so feel free to ignore.